Gems/Prefab/PrefabDependencyViewer/Code/Source/MainWindow.h

```diff
@@ -0,0 +1,139 @@
+
+        ////////////////// PrefabDependencyViewerWidget overrides ////////////////////
+        virtual void DisplayTree(const Utils::DirectedGraph& graph) override;
+
+        void DisplayNodesByLevel(const Utils::DirectedGraph& graph, AZStd::vector<int> numNodesAtEachLevel, int widestLevel);
+
+        void DisplayNode(Utils::Node* node, AZ::Vector2 pos);
+
+        
+        GraphCanvas::SlotId CreateExecutionSlot(
+            GraphCanvas::NodeId nodeId,
+            const AZStd::string& slotName,
+            const AZStd::string& tooltip,
+            GraphCanvas::SlotGroup slotGroup,
+            bool isInput);
+
+        GraphCanvas::SlotId CreateDataSlot(
+            GraphCanvas::NodeId nodeId,
+            const AZStd::string& slotName,
+            const AZStd::string& tooltip,
+            AZ::Uuid dataType,
+            GraphCanvas::SlotGroup slotGroup,
+            bool isInput);
+
+        void AddSlotToNode(AZ::Entity* slotEntity, GraphCanvas::NodeId nodeId);
+        void CreateNodeUi(const AzToolsFramework::Prefab::TemplateId& tid);
+
+    protected:
+        ////////////////// GraphCanvas::AssetEditorMainWindow overrides //////////////
+        /** Overriding RefreshMenu in order to remove the
+        unnecessary menu bar on the top. As a bonus, this
+        also removes the ability to revive NodePalette from the UI. */
+        virtual void RefreshMenu() override {}
+
+        // GraphCanvas::GraphModelRequestBus::Handler overrides
+        virtual void RequestUndoPoint() {}
+        virtual void RequestPushPreventUndoStateUpdate() {}
+        virtual void RequestPopPreventUndoStateUpdate() {}
+        virtual void TriggerUndo() {}
+        virtual void TriggerRedo() {}
+        //! This is sent when a connection is disconnected.
+        virtual void DisconnectConnection([[maybe_unused]]const ConnectionId& connectionId)
+        {
+        }
+        //! This is sent when attempting to create a given connection.
+        virtual bool CreateConnection(
+            [[maybe_unused]] const ConnectionId& connectionId,
+            [[maybe_unused]] const Endpoint& sourcePoint,
+            [[maybe_unused]] const Endpoint& targetPoint)
+        {
+            return true;
+        }
+        //! This is sent to confirm whether or not a connection can take place.
+        virtual bool IsValidConnection([[maybe_unused]] const Endpoint& sourcePoint, [[maybe_unused]] const Endpoint& targetPoint) const
+        {
+            return true;
+        }
+        //! This is sent to confirm whether or not a variable assignment can take place.
+        virtual bool IsValidVariableAssignment([[maybe_unused]]const AZ::EntityId& variableId, [[maybe_unused]]const Endpoint& targetPoint) const
+        {
+            return true;
+        }
+
+                //! Get the Display Type name for the given AZ type
+        virtual AZStd::string GetDataTypeString([[maybe_unused]] const AZ::Uuid& typeId)
+        {
+            return "";
```

Comment:
Nitpick: Instead of an empty string return a default constructed string. This safes a little bit on having to initialize a default `AZStd::string` to the same values as the provided string is empty.

Reply:
Resolved